### PR TITLE
Refactor tx status waiter

### PR DESCRIPTION
**Pull request by dolgopolovwork**
### Description of the Change
Refactored transaction status waiter. Now it stores tx status(COMMITTED, REJECTED, etc) as well.

### CODE REVIEW

File: notary-commons/src/main/kotlin/com/d3/commons/sidechain/iroha/consumer/status/TxStatus.kt [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+         * @return failed status
+         */
+        fun createFailed(state: TxState, ex: Exception) = TxStatus(state, ex)
+    }
+}
+
+/**
+ * Possible states of transactions
+ */
+enum class TxState {
```
Comment (Mingela) [inline notary-commons/src/main/kotlin/com/d3/commons/sidechain/iroha/consumer/status/TxStatus.kt]: why do you need your own enum? there is `TxStatus` enum in IPJ

File: notary-commons/src/main/kotlin/com/d3/commons/sidechain/iroha/consumer/status/TxStatus.kt [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        }
+        .onMstExpired { expiredTx ->
+            statusReference.set(
+                TxStatus.createFailed(
+                    TxState.MST_EXPIRED,
+                    TimeoutException("Tx ${expiredTx.txHash} MST expired. ${expiredTx.errOrCmdName}")
+                )
+            )
+        }
+        .onNotReceived { failedTx ->
```
Comment (Mingela) [inline notary-commons/src/main/kotlin/com/d3/commons/sidechain/iroha/consumer/status/TxStatus.kt]: do we really need this processing?

File: notary-commons/src/main/kotlin/com/d3/commons/sidechain/iroha/consumer/status/TxStatus.kt [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        }
+        .onRejected { rejectedTx ->
+            statusReference.set(
+                TxStatus.createFailed(
+                    TxState.REJECTED,
+                    IOException("Tx ${rejectedTx.txHash} was rejected. ${rejectedTx.errOrCmdName}")
+                )
+            )
+        }
+        .onTransactionFailed { failedTx ->
```
Comment (Mingela) [inline notary-commons/src/main/kotlin/com/d3/commons/sidechain/iroha/consumer/status/TxStatus.kt]: What does `Failed` mean from Iroha perspective?

Comment (dolgopolovwork) [reply notary-commons/src/main/kotlin/com/d3/commons/sidechain/iroha/consumer/status/TxStatus.kt]: Probably stateless and stateful errors

File: notary-commons/src/main/kotlin/com/d3/commons/sidechain/iroha/consumer/status/TxStatus.kt (hunk 2 above)
Comment (dolgopolovwork) [reply notary-commons/src/main/kotlin/com/d3/commons/sidechain/iroha/consumer/status/TxStatus.kt]: Are we talking about `onNotReceived`? If it so, yes.

File: notary-commons/src/main/kotlin/com/d3/commons/sidechain/iroha/consumer/status/TxStatus.kt (hunk 1 above)
Comment (dolgopolovwork) [reply notary-commons/src/main/kotlin/com/d3/commons/sidechain/iroha/consumer/status/TxStatus.kt]: this is a good point.

**Review by Mingela — APPROVED** [review]

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**